web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/vedtak/aksjonspunkt/AbstractVedtaksbrevOverstyringshåndterer.java

```diff
@@ -47,82 +45,103 @@ public abstract class AbstractVedtaksbrevOverstyringshåndterer {
     }
 
     OppdateringResultat håndter(VedtaksbrevOverstyringDto dto, AksjonspunktOppdaterParameter param, boolean toTrinn) {
-        var begrunnelse = dto.getBegrunnelse();
-        var behandling = getBehandling(param.getBehandlingId());
+        var behandlingId = param.getBehandlingId();
+        var behandling = behandlingRepository.hentBehandling(behandlingId);
+        oppdaterVedtaksbrev(dto, behandling);
+        opprettHistorikkinnslag(param.getRef(), behandling, toTrinn, dto.isSkalBrukeOverstyrendeFritekstBrev());
 
-        oppdaterBegrunnelse(behandling, begrunnelse);
-        var builder = OppdateringResultat.utenTransisjon()
-            .medTotrinnHvis(dto.isSkalBrukeOverstyrendeFritekstBrev());
-        if (dto.isSkalBrukeOverstyrendeFritekstBrev()) {
-            oppdaterFritekstVedtaksbrev(dto, param);
-        } else {
-            fjernFritekstBrevHvisEksisterer(behandling);
-        }
-        opprettHistorikkinnslag(param.getRef(), behandling, toTrinn);
+        var builder = OppdateringResultat.utenTransisjon();
         if (toTrinn) {
             opprettAksjonspunktForFatterVedtak(behandling, builder);
+            builder.medTotrinnHvis(true);
```

Comment:
Hm, her gjør du 5015 til totrinn uansett (kaller håndter(... true)) - mens 5028 er avhengig av valget i dto.
Aksjonspunktet løses med ForeslåVedtakAksjonspunktDto når når du trykker Send til beslutter 
5015 er i utgangspunktet 1-trinn og skal bli 2trinn dersom man har redigert brevet - ellers ikke.
Det er gjerne et annet aksjonspunkt som gjør at man får 5015 (totrinn) framfor 5028. Men selve aksjonspunktet 5015 skal bare kunne returneres dersom beslutter vil ha endringer i brevet. 
Resten trenger vi gå gjennom sammen.

Reply:
Ja, det stemmer. Denne har jeg lyst til å diskutere litt fordi jeg er ikke helt forstått med hva forskjellen på 
OppdateringResultat.totrinnsKontroll og
Behandling.toTrinnsBehandling

For her setter alltid OppdateringResultat.totrinnsKontroll til det boolean fra DTO
Mens Behandling.toTrinnsBehandling settes alltid til true ved 5015.

Så hva betyr det at OppdateringResultat.totrinnsKontroll=false, mens Behandling.toTrinnsBehandling=true?

Reply:
Gammel logikk var da følgende:
```
        var builder = OppdateringResultat.utenTransisjon()
            .medTotrinnHvis(dto.isSkalBrukeOverstyrendeFritekstBrev());
        if (toTrinn) {
            opprettAksjonspunktForFatterVedtak(behandling, builder);
            behandling.setToTrinnsBehandling();
        } else {
            behandling.nullstillToTrinnsBehandling();
        }
```

Reply:
Behandling.isToTrinn er avledet - settes litt intrikat. Det at det finnes 5015 betyr at det kommer 5016 - beslutte - dvs totrinns.

Aksjonspunkt.isToTrinn er utledet dels av AksjonspunktDef, dels av ApOppdaterer. Det er bare AP.isTotrinn som skal vurderes av beslutter.

5015 og 5028 er i utgangspunktet entrinns-ap. Det skal bare få satt totrinns dersom man overstyrer brevet. Det er slik det er i dag og det gir ikke mening at 5015 skal vurderes av beslutter uten overstyrt brev.

AksjonspunktResultat brukes til mye - først og fremst å sette Aksjonspunkt.isToTrinn. Behandling er sekundert.

Forskjeller kan oppstå dersom Ap er totrinns første gang hos beslutter, men at det avbrytes ifm senere behandling og dermed ikke skal til vurdering.

Reply:
Endringen jeg gjorde, som mulig ikke er korrekt er:
```
        var builder = OppdateringResultat.utenTransisjon():
        if (toTrinn) {
            opprettAksjonspunktForFatterVedtak(behandling, builder);
            builder..medTotrinnHvis(true);
            behandling.setToTrinnsBehandling();
        } else {
            behandling.nullstillToTrinnsBehandling();
        }
```

Reply:
Skjønner. Da flytter jeg denne ut og setter builder.medTotinnhvis() basert på om det overstyres eller ei.

Reply:
Da har jeg oppdatert til

```
        var builder = OppdateringResultat.utenTransisjon()
            .medTotrinnHvis(dto.isSkalBrukeOverstyrendeFritekstBrev());
        if (toTrinn) {
            opprettAksjonspunktForFatterVedtak(behandling, builder);
            behandling.setToTrinnsBehandling();
        } else {
            behandling.nullstillToTrinnsBehandling();
        }
```

Og lagt til et par nye tester.